packages/engine/tests/simulations/mod.rs

```diff
@@ -0,0 +1,75 @@
+mod experiment;
+
+/// Opens `$project` relatively to this file as HASH project and uses
+/// `$project/integration-test.json` as configuration for the test.
+///
+/// The configuration contains a list of objects, where each object has the following values:
+/// - for a simple experiment:
+///   - "experiment": Name of the experiment to run as specified in `experiments.json`
+///   - "expected-outputs": List of expected outputs, where the length of the list must be equal to
+///     the number of simulations of the experiment. Each entry of the list is an objects with the
+///     step number as key mapping to the expected outputs of the corresponding step in the
+///     simulation run:
+///     - "json-state": subset of the values expected in the `json_state.json` output
+///     - "globals": subset of the values expected in the `globals.json` output
+///     - "analysis-outputs": subset of the values expected in the `analysis_outputs.json` output
+/// - for single-run experiments:
+///   - "steps": Number of steps to run
+///   - "expected-output": Expected output of the simulation containing with the same schema as one
+///     element in "expected-outputs"
+macro_rules! run_test {
+    ($project:tt) => {
+        #[test]
+        fn $project() {
+            use $crate::simulations::experiment::*;
+
+            let project_path = std::path::Path::new(file!())
```

Comment:
Is there much of a benefit to doing this? 
We've sort of locked ourselves into relative paths at the moment, which isn't ideal but that's how it is for now. So, considering the engine has to be ran from the package root, why don't we just define integration tests as relative paths from that (like we do with all other paths)?

Reply:
It's not a big deal to simply use the current file path, so I don't see why we shouldn't do it correctly.

Reply:
Well I'm not sure that the method here is necessarily doing it "correctly". `file!()` returns a relative path from the root anyway, right? It's an extra level of indirection, that's forcing a relation between the location of this file, the package root, and the simulation project. Whereas using relative paths from the root means you only have one place to update if you move the experiment, doesn't matter where this file is relative to it.

Reply:
I updated it to use hardcoded relative paths